mbed_build/_internal/templates/mbed_config.tmpl

```diff
@@ -1,28 +1,63 @@
-/*
-* Copyright (C) 2020 Arm Mbed. All rights reserved.
-* SPDX-License-Identifier: Apache-2.0
-*/
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
 
-// Automatically generated configuration file.
-// DO NOT EDIT, content will be overwritten.
+# Automatically generated configuration file.
+# DO NOT EDIT, content will be overwritten.
+add_library(gen_config INTERFACE)
+set(MBED_TOOLCHAIN "{{toolchain_name}}" CACHE STRING "")
+set(MBED_TARGET "{{target_name}}" CACHE STRING "")
+set(MBED_CPU_CORE "{{core}}" CACHE STRING "")
 
-#ifndef __MBED_CONFIG_H__
-#define __MBED_CONFIG_H__
+if(MBED_TOOLCHAIN STREQUAL "GCC_ARM")
+    add_definitions(-DTOOLCHAIN_GCC_ARM -DTOOLCHAIN_GCC)
+elseif(MBED_TOOLCHAIN STREQUAL "ARM")
+    add_definitions(-DTOOLCHAIN_ARM)
+endif()
```

Comment:
These definitions should come from the per-toolchain cmake files instead. We just want to create CMake variables here to expose the Mbed configuration to CMake, rather than interface directly with the toolchain.